src/platform_impl/linux/wayland/window/state.rs

```diff
@@ -726,10 +726,22 @@ impl WindowState {
     }
 
     /// Set the custom cursor icon.
-    pub(crate) fn set_custom_cursor(&mut self, cursor: &CursorImage) {
+    pub(crate) fn set_custom_cursor(&mut self, cursor: RootCustomCursor) {
+        let cursor = match cursor {
+            RootCustomCursor {
+                inner: PlatformCustomCursor::Wayland(cursor),
+            } => cursor.0,
+            RootCustomCursor {
+                inner: PlatformCustomCursor::X11(_),
+            } => {
+                error!("passed a X11 cursor to Wayland backend");
```

Comment:
I could change those to `unreachable!`, because you can't recreate event loops on those platforms, right?